packages/flutter/lib/src/material/shaders/stretch_overscroll.frag

```diff
@@ -0,0 +1,161 @@
+  float u_stretch_affected_dist,
+  float u_inverse_stretch_affected_dist,
+  float distance_stretched,
+  float interpolation_strength
+) {
+  float offset_pos = u_stretch_affected_dist - in_pos;
+  float pos_based_variation = mix(
+    1.0,
+    ease_in(offset_pos, u_inverse_stretch_affected_dist),
+    interpolation_strength
+  );
+  float stretch_intensity = overscroll * pos_based_variation;
+  return distance_stretched - (offset_pos / (1.0 + stretch_intensity));
+}
+
+float compute_overscroll_end(
+  float in_pos,
+  float overscroll,
+  float reverse_stretch_dist,
+  float u_stretch_affected_dist,
+  float u_inverse_stretch_affected_dist,
+  float distance_stretched,
+  float interpolation_strength,
+  float viewport_dimension
+) {
+  float offset_pos = in_pos - reverse_stretch_dist;
+  float pos_based_variation = mix(
+    1.0,
+    ease_in(offset_pos, u_inverse_stretch_affected_dist),
+    interpolation_strength
+  );
+  float stretch_intensity = (-overscroll) * pos_based_variation;
+  return viewport_dimension - (distance_stretched - (offset_pos / (1.0 + stretch_intensity)));
+}
+
+float compute_streched_effect(
+  float in_pos,
+  float overscroll,
+  float u_stretch_affected_dist,
+  float u_inverse_stretch_affected_dist,
+  float distance_stretched,
+  float distance_diff,
+  float interpolation_strength,
+  float viewport_dimension
+) {
+  if (overscroll > 0.0) {
+    if (in_pos <= u_stretch_affected_dist) {
+      return compute_overscroll_start(
+        in_pos, overscroll, u_stretch_affected_dist,
+        u_inverse_stretch_affected_dist, distance_stretched,
+        interpolation_strength
+      );
+    } else {
+      return distance_diff + in_pos;
+    }
+  } else if (overscroll < 0.0) {
+    float stretch_affected_dist_calc = viewport_dimension - u_stretch_affected_dist;
+    if (in_pos >= stretch_affected_dist_calc) {
+      return compute_overscroll_end(
+        in_pos,
+        overscroll,
+        stretch_affected_dist_calc,
+        u_stretch_affected_dist,
+        u_inverse_stretch_affected_dist,
+        distance_stretched,
+        interpolation_strength,
+        viewport_dimension
+      );
+    } else {
+      return -distance_diff + in_pos;
+    }
+  } else {
+    return in_pos;
+  }
+}
+
+out vec4 frag_color;
+
+void main() {
+  vec2 uv = FlutterFragCoord().xy / u_size;
+  float in_u_norm = uv.x;
+  float in_v_norm = uv.y;
+
+  float out_u_norm;
+  float out_v_norm;
+
+  float norm_stretch_affected_dist_x = 1.0;
+  float norm_stretch_affected_dist_y = 1.0;
+
+  float norm_inverse_stretch_affected_dist_x = 1.0;
+  float norm_inverse_stretch_affected_dist_y = 1.0;
+
+  float norm_distance_stretched_x = 1.0 / (1.0 + abs(u_overscroll_x));
+  float norm_distance_stretched_y = 1.0 / (1.0 + abs(u_overscroll_y));
+
+  float norm_dist_diff_x = norm_distance_stretched_x - 1.0;
+  float norm_dist_diff_y = norm_distance_stretched_y - 1.0;
+
+  float norm_viewport_width = 1.0;
+  float norm_viewport_height = 1.0;
```

Comment:
Can this part be simplified using vec computation?

```

// Global
const vec2 norm_stretch_affected_dist(1.0, 1.0);
const vec2 norm_inverse_stretch_affected_dis(1.0, 1.0);
uniform vec2 u_overscroll;

  vec2 in_uv_norm = FlutterFragCoord().xy / u_size;
  vec2 out_uv_norm;
  vec2 norm_distance_stretched = 1.0 / (1.0 + abs(u_overscroll));
  vec2 norm_dist_diff = norm_distance_stretched - vec2(1.0, 1.0);
  vec2 norm_viewport_size(1.0, 1.0);
```

Reply:
this is only relevant though if we need to stretch both x/y. If we on;y ever need to stretch in one dimension then we should leave it scalar math and just apply it in one direction

Reply:
Ah I see. Then this problem is equivalent to the comment below. I'll resolve this one then.